language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageServer.java

```diff
@@ -0,0 +1,170 @@
+      Path jarPath;
+      try {
+          // Compute the absolute jar path based on the workspace folder, if a relative path is specified.
+          jarPath = workspaceFolders != null && workspaceFolders.size() > 0?
+              Paths.get(new URI(initializeParams.getWorkspaceFolders().get(0).getUri()).getPath()).resolve(compilerJar) :
+              Paths.get(compilerJar);
+      } catch (URISyntaxException e) {
+        throw new RuntimeException(e);
+      }
+      loader = Util.getJarClassLoader(jarPath);
+    }
+
+    // Initialize Oberon0 grammars
+    // Includes parser & driver
+    // TODO: should the init grammar be specifiable?
+    try {
+        Util.initGrammar("edu:umn:cs:melt:Oberon0:artifacts:A5", loader);
+    } catch (SecurityException | ReflectiveOperationException e) {
+        client.showMessage(new MessageParams(MessageType.Error, "Error loading compiler jar " + compilerJar + ": " + e.toString()));
+    }
+
+    // Load the specified parser
+    boolean loadedParser = false;
+    try {
+        service.setParserFactory(Util.loadCopperParserFactory(loader, parserName, NModule_c.class));
+        loadedParser = true;
+    } catch (SecurityException | ReflectiveOperationException e) {
+        client.showMessage(new MessageParams(MessageType.Error, "Error loading parser " + parserName + " from jar: " + e.toString()));
+    }
+
+    // Initialize the project folder(s)
+    if (initializeParams.getWorkspaceFolders() != null) {
+        service.setWorkspaceFolders(initializeParams.getWorkspaceFolders());
+    }
+
+    // Set the capabilities of the LS to inform the client.
+    ServerCapabilities capabilities = new ServerCapabilities();
+    capabilities.setTextDocumentSync(TextDocumentSyncKind.Full);
+    if (loadedParser) {
+        capabilities.setSemanticTokensProvider(
+            new SemanticTokensWithRegistrationOptions(
+                new SemanticTokensLegend(
+                    Oberon0LanguageService.tokenTypes, Oberon0LanguageService.tokenModifiers),
+                new SemanticTokensServerFull(false), false));
+    }
+
+    capabilities.setDeclarationProvider(true);
+
+    FileOperationOptions fileOperationOptions = new FileOperationOptions(
+        List.of(new FileOperationFilter(new FileOperationPattern("**/*.ob")))
+    );
+    FileOperationsServerCapabilities fileOperations = new FileOperationsServerCapabilities();
+    fileOperations.setDidCreate(fileOperationOptions);
+    fileOperations.setDidDelete(fileOperationOptions);
+    fileOperations.setDidRename(fileOperationOptions);
+    WorkspaceServerCapabilities workspaceServer = new WorkspaceServerCapabilities();
+    workspaceServer.setFileOperations(fileOperations);
+    capabilities.setWorkspace(workspaceServer);
+
+    final InitializeResult initializeResult = new InitializeResult(capabilities);
+    return CompletableFuture.supplyAsync(()->initializeResult);
+
+	}
```

Comment:
Indentation is off here and on lines below.

Reply:
Done in [805e115](https://github.com/melt-umn/Oberon0/pull/14/commits/805e1155bddf92a6c4eabd0f412a8f6a6222e774)